### DISCUSSION TITLE
Why is auto-materialisation launching occasional duplicated runs

### DISCUSSION THREAD

**Discussion by bolinzzz** [dagster-io/dagster#18432] [Q&A] [answered]
Hi all! My team deployed Dagster to K8s and is using auto-materialisation. Lately we noticed that there are occasional duplicated runs. See the scrrenshot below of an asset from our production environment:
![Screenshot 2023-12-01 at 11 40 29 PM](https://github.com/dagster-io/dagster/assets/101250079/5729ea15-cdba-48e1-873d-99cb196b8ace)

On 29 and 30 Nov, two auto-materialisations or runs happened right after one another. For other days, only one run was submitted. 
Here's the auto-materialisation history
![Screenshot 2023-12-01 at 11 41 28 PM](https://github.com/dagster-io/dagster/assets/101250079/03c49e61-7e8b-479b-b5ef-484ad1479c72)

Just wondering what might be causing this? Could it be K8s evicting and moving dagster-run pods from one node to another when they just started? Please provide some insight and guidance!

**Comment by bolinzzz** [comment]
I think I know the reason. Looking at 30 Nov. The auto-materialisation first launched the run at 13:00, but the run did not start until 13:09. Attached a screenshot of the run events 
![Screenshot 2023-12-01 at 11 52 45 PM](https://github.com/dagster-io/dagster/assets/101250079/afa17e3d-4493-48a8-9a31-8cbd8588a8f3)


I guess the auto-materialisation evaluation happens every 10 minutes, so at 13:10, the asset freshness policy is still violated thus another run is launched. For the second run, it is able to start right away
![Screenshot 2023-12-01 at 11 55 27 PM](https://github.com/dagster-io/dagster/assets/101250079/4ec4cda8-d4ee-4f0b-b650-53bd0576ad46)

**Reply by OwenKephart** [reply]
Hi @bolinzzz -- in theory, the evaluation should take into account runs that have not started / are in the queue when deciding what to kick off, so this seems like a bug to me.

Do you have some more information on your setup? What freshness policies you have, the shape of your graph, etc.?

**Reply by bolinzzz** [reply]
Hi Owen, we are on Dagster 1.5.9. The auto-materialisation policy of the asset is 
Lazy with the following freshness policy `By 01:30 UTC, this asset should incorporate all data up to 1 day before that time.`. We will be adding`AutoMaterializeRule.materialize_on_cron`. 


A screenshot of the graph:
![Screenshot 2023-12-04 at 10 26 31 AM](https://github.com/dagster-io/dagster/assets/101250079/e129a486-d8cd-4383-b2bf-9ea81276ad20)

[a comment by an automated account is not shown]

**Comment by cyberjar09** [comment]
we are on `1.7.8` and still seeing the issue

**Reply by garethbrickman** [reply]
As of 1.8.0, Auto-Materialization is replaced by Declarative Automation. More info in https://github.com/dagster-io/dagster/discussions/22811

**Reply by bolinzzz** [reply, marked as answer]
`AutoMaterializeRule.skip_on_run_in_progress()` which has been added as part of `1.6.12` release, it is not part of `AutoMaterializePolicy.eager()` by default.


See this: https://github.com/dagster-io/dagster/discussions/19214
